### PR TITLE
test: increase unit test coverage in /pkg/services/sso

### PR DESCRIPTION
**Pull request by JasonPowr**
## Description
Increase unit test coverage in /pkg/services/sso

| File  | Coverage Before | Coverage After |
|---|---|---|
| redhatsso_service.go  |  24.7% | 93.8% |
|  keycloak_service.go | 41%  | 87.1% |
|  keycloak_service_builder.go | 34.3%  | 83.3% |

## Verification Steps
1. Run `make test`
2. Run `make test/html/coverage/report`
3. Should be able to see the increased test coverage

## Checklist (Definition of Done)

- [x] All acceptance criteria specified in JIRA have been completed
- [x] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] ~~Documentation added for the feature~~
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
- [ ] ~~Required metrics/dashboards/alerts have been added (or PR created).~~
- [ ] ~~Required Standard Operating Procedure (SOP) is added.~~
- [ ] ~~JIRA has been created for changes required on the client side~~

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -334,6 +334,8 @@ test: gotestsum
     -e "github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/services/account/accountservice_interface.go" \
     -e "github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/services/account/organization_type.go" \
     -e "github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/services/account/account_mock.go" \
+	-e "github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/services/sso/keycloak.go" \
+	-e "github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/services/sso/keycloak_service_proxy.go" \
```
Comment (machi1990) [inline Makefile:338]: Can we align this?
```suggestion
-e "github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/services/sso/keycloak.go" \
-e "github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/services/sso/keycloak_service_proxy.go" \
```

Comment (JasonPowr) [reply Makefile:338]: Hey @machi1990 Im not sure what is going on here on my end it is aligned. Any suggestions?

Comment (machi1990) [reply Makefile:338]: I think the issue is the usage of `tabs` instead of spaces as for the others.

Comment (JasonPowr) [reply Makefile]: yep you were right should be fixed now :)

**Comment by machi1990** [comment]
The change lgtm

I've asked @akoserwal and @ziccardi review on this as well as @pawelpaszki

**Review by pawelpaszki — APPROVED** [review]

**Review by ziccardi — COMMENTED** [review]
The change LGTM.
I'm just not sure it makes much sense unit test the small pieces composing the builder by itself.

I think it would make more sense to test unit test the whole builder by testing that:
```go
sso.NewKeycloakServiceBuilder().
				ForOSD().
				WithConfiguration(testCond).
				WithRealmConfig(testRealmConf).
				Build()
```
and 
```go
return sso.NewKeycloakServiceBuilder().
				ForKFM().
				WithConfiguration(testConf).
				Build()
```
are returning what is expected.

WDYT?

File: pkg/services/sso/keycloak_service_builder_test.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+		},
+	}
+	g := NewWithT(t)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			builder := &osdKeycloackServiceBuilder{
+				config:      tt.fields.config,
+				realmConfig: tt.fields.realmConfig,
+			}
+			g.Expect(builder.Build()).To(Not(BeNil()))
```
Comment (ziccardi) [inline pkg/services/sso/keycloak_service_builder_test.go:69]:
```suggestion
service :=  NewKeycloakServiceBuilder().
                        ForOSD().
                        WithConfiguration(&tt.fields.config).
                        WithRealmConfig(&tt.fields.realmConfig).
                        Build()
Expect(service).ToBe(Equal(tt.want))                     
```

File: pkg/services/sso/keycloak_service_builder_test.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+	}{
+		{
+			name: "should return an instance of the OSD keycloak service",
+			fields: fields{
+				config: &keycloak.KeycloakConfig{
+					SelectSSOProvider: keycloak.REDHAT_SSO,
+				},
+				realmConfig: &keycloak.KeycloakRealmConfig{},
+			},
+			want: NewKeycloakServiceBuilder().ForOSD().WithConfiguration(&keycloak.KeycloakConfig{}).WithRealmConfig(&keycloak.KeycloakRealmConfig{}).Build(),
```
Comment (ziccardi) [inline pkg/services/sso/keycloak_service_builder_test.go:59]: we are testing the builder, so `want` should be independent of the builder.

File: pkg/services/sso/keycloak_service_builder_test.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+	}
+	g := NewWithT(t)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			service := NewKeycloakServiceBuilder().
+				ForKFM().
+				WithConfiguration(tt.fields.config).
+				WithRealmConfig(tt.fields.realmConfig).
+				Build()
+			g.Expect(service).To(Not(BeNil()))
```
[the remark this answers is not in the document]
Comment (ziccardi) [reply pkg/services/sso/keycloak_service_builder_test.go:36]: I believe it should be at least:
```suggestion
			g.Expect(service).ToNot(BeNil())
			g.Expect(service.GetConfig()).To(Equal(tt.want.GetConfig()))
			g.Expect(service.GetRealmConfig()).To(Equal(tt.want.GetRealmConfig()))
```

and tt.want should be correctly initialised. Ex:
```go
			want: &keycloakServiceProxy{
				service: &redhatssoService{
					client: redhatsso.NewSSOClient(&keycloak.KeycloakConfig{
						SelectSSOProvider: keycloak.MAS_SSO,
					}, &keycloak.KeycloakRealmConfig{}),
				},
			},
```

File: pkg/services/sso/keycloak_service_builder_test.go [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+	tests := []struct {
+		name   string
+		fields fields
+		want   KeycloakService
+	}{
+		{
+			name: "should return an instance of the keycloak service",
+			fields: fields{
+				config:      &keycloak.KeycloakConfig{},
+				realmConfig: &keycloak.KeycloakRealmConfig{},
```
Comment (ziccardi) [inline pkg/services/sso/keycloak_service_builder_test.go:25]: Better to add some value into the configs, so we are sure they are being used,

**Pull request merged**